earth_enterprise/src/common/khTypes.h

```diff
@@ -19,6 +19,26 @@
 #define GEO_EARTH_ENTERPRISE_SRC_COMMON_KHTYPES_H_
 
 #include <assert.h>
+#include <mutex>
+#include <vector>
+
+template <typename T>
```

Comment:
The class below is really only protecting a pretty small section of code because vector's push_back isn't virtual and a lot of vector references are passed around.  If you change MTVector to private inheritance you'll start seeing all of these because nothing will compile.